### PR TITLE
SLING-8421: Allow artifact providers that work with URLs instead of F…

### PR DESCRIPTION
**Pull request by karlpauls**
…iles.

### CODE REVIEW

**Comment by bosschaert** [comment]
The reason that the build failed is that this commit depends on Feature Model 1.0.2 artifacts which were under vote when the CI build was executed. 
The artifacts are released now, and should appear in Maven Central shortly. @rombert is there a was to trigger the CI build again?

**Comment by rombert** [comment]
@bosschaert - click the 'details' link thext to the failed check. If logged in, there should be a 'Rerun' button in the top bar.

**Comment by bosschaert** [comment]
+1 (assuming Jenkins is green)

Ps @rombert thanks for the tip on rebuilding.

**Comment by karlpauls** [comment]
Jenkins can't be green for this one at this point because it requires https://github.com/apache/sling-org-apache-sling-feature-io/pull/12 to work. We'd have to merge both at the same time.

File: src/main/java/org/apache/sling/feature/launcher/impl/Installation.java [hunk 1]
```diff
@@ -115,6 +136,12 @@ public void addFrameworkProperty(String key, String value)
      */
     @Override
     public Map<Integer, List<File>> getBundleMap() {
+        return convert(this.bundleMap);
+    }
+
+    @Override
+    public Map<Integer, List<ArtifactHandler>> getBundleArtifactMap()
+    {
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/feature/launcher/impl/Installation.java]: Can we remove getBundleMap() ?

File: src/main/java/org/apache/sling/feature/launcher/impl/Installation.java [hunk 2]
```diff
@@ -131,6 +158,12 @@ public void addFrameworkProperty(String key, String value)
      */
     @Override
     public List<File> getInstallableArtifacts() {
+        return convertHandlers(this.installables);
+    }
+
+    @Override
+    public List<ArtifactHandler> getInstallableArtifactHandlers()
+    {
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/feature/launcher/impl/Installation.java]: Can we remove getInstallableArtifacts?

File: src/main/java/org/apache/sling/feature/launcher/spi/LauncherRunContext.java [hunk 3]
```diff
@@ -58,6 +66,12 @@
      */
     List<File> getInstallableArtifacts();
 
+    /**
+     * List of installable artifacts.
+     * @return The list of files. The list might be empty.
+     */
+    List<ArtifactHandler> getInstallableArtifactHandlers();
+
     /**
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/feature/launcher/spi/LauncherRunContext.java]: Changes to the spi increase the package version

**Pull request closed without merging**